inductor: reduce complie time by reducing repr calls of quantize or Opaque tensor

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104696


For quantize or opaue tensor, if they are constant values, the calls of  tensor ```__repr__``` will have memory copy(https://github.com/pytorch/pytorch/blob/main/torch/_tensor_str.py#L550):
https://github.com/pytorch/pytorch/blob/db1ac4e29b0f557711190c8d49d4afb5da1844e8/torch/_inductor/codegen/wrapper.py#L289-L292

for CPP codegen, there have many times of initiation of ```WrapperCodeGen```: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/codegen/cpp.py#L2023, which consumes much time.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng

XiaobingSuper committed:
inductor: reduce complie time by reducing repr calls of quantize or Opaque tensor

[ghstack-poisoned]

desertfire commented on torch/_inductor/codegen/wrapper.py:
```diff
@@ -286,9 +286,9 @@ def __init__(self):
+        for name, value in V.graph.constant_reprs.items():
             # include a hash so our code cache gives different constants different files
-            hashed = hashlib.sha256(repr(value).encode("utf-8")).hexdigest()
+            hashed = hashlib.sha256(value).hexdigest()
```
Should we go a step further by storing {name:hashed} ?

XiaobingSuper committed:
Update on "inductor: reduce complie time by reducing repr calls of quantize or Opaque tensor"



For quantize or opaue tensor, if they are constant values, the calls of  tensor ```__repr__``` will have memory copy(https://github.com/pytorch/pytorch/blob/main/torch/_tensor_str.py#L550):
https://github.com/pytorch/pytorch/blob/db1ac4e29b0f557711190c8d49d4afb5da1844e8/torch/_inductor/codegen/wrapper.py#L289-L292

for CPP codegen, there have many times of initiation of ```WrapperCodeGen``: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/codegen/cpp.py#L2023, which consumes many time.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng

[ghstack-poisoned]

XiaobingSuper commented:
@pytorchbot merge
